src/queries.ts

```diff
@@ -0,0 +1,128 @@
+import { GQLEdgeInterface, GQLTransactionsResultInterface } from './gql_types';
+import fetch from 'node-fetch';
+
+const GQL_URL = 'https://arweave.net/graphql';
+const ITEMS_PER_REQUEST = 100;
+const VALID_APP_NAMES = ['ArDrive-Web', 'ArDrive-CLI', 'ArDrive-Sync'] as const;
+
+interface Query {
+	query: string;
+}
+
+export interface StakedPSTHolders {
+	[address: string]: number;
+}
+
+export async function getWaleltsEligibleForStreak(): Promise<StakedPSTHolders> {
+	return getStakedPSTHolders()
+		.then((result) => Object.entries(result))
+		.then((entries) => entries.filter((data) => data[1] >= 200))
+		.then((entries) => Object.fromEntries(entries));
+}
+
+async function getStakedPSTHolders(): Promise<StakedPSTHolders> {
+	// https://v2.cache.verto.exchange/-8A6RexFkpfWwuyVO98wzSFZh0d6VJuI-buTJvlwOJQ
+	// const result = await vertoClient.getExchangeDetails('ARDRIVE');
+	const blockHeightRequest = await fetch('https://arweave.net/height');
+	const blockHeight = +(await (await blockHeightRequest.blob()).text());
+	const result = await fetch('https://v2.cache.verto.exchange/-8A6RexFkpfWwuyVO98wzSFZh0d6VJuI-buTJvlwOJQ');
```

Comment:
maybe a better name like `pstRequest`?
I was thinking now, why verto and not arweave gw? I know that we use verto on the price calculator. I'm just curious.

Reply:
484898176e4e57673d28735fe5f928b9e44ec391

Reply:
> why verto and not arweave gw?

Let's better discuss it with the team